api/v1alpha1/provisioning_types.go

```diff
@@ -21,6 +21,16 @@ import (
 	operatorv1 "github.com/openshift/api/operator/v1"
 )
 
+// ProvisioningNetwork is the boot mode of the system
+// +kubebuilder:validation:Enum=Managed;Unmanaged;Disabled
+type ProvisioningNetwork string
+
+const (
+	provisioningNetworkManaged   = "Managed"
```

Comment:
These consts can be defined as being of the new type. I think if you do that, you don't need line 25.

Reply:
I removed these constants from the latest changes because they are not necessary to generate the right yaml file. Will add them as part of the validation code that is in review.